src/main/java/frc/robot/classes/spikemotor/SpikeMotor.java

```diff
@@ -0,0 +1,252 @@
+                "Motor must be initialized first"
+            );
+        for (SpikeMotor follower : followers) {
+            follower.moveTo(position);
+        }
+        moveToImpl(position);
+    }
+
+    /**
+    * The implementation of init.
+    *
+    * @param deviceNumber    an int giving the device number
+    * @return                nothing
+    * @see                   #init(int deviceNumber)
+    */
+    abstract protected void initImpl(int deviceNumber);
+
+    /**
+    * The implementation of setSpeed.
+    * @param speed    a double indicating the speed
+    * @return         nothing
+    * @see            #setSpeed(double speed)
+    */
+    abstract protected void setSpeedImpl(double speed);
+
+    /**
+    * The implementation of getSpeed.
+    *
+    * @return    a double indicating the speed
+    * @see       #getSpeed()
+    */
+    abstract protected double getSpeedImpl();
+
+    /**
+    * The implementation of setPosition.
+    *
+    * @param position   a double indicating the position to set
+    * @return           nothing
+    * @see              #setPosition()
+    */
+    abstract protected void setPositionImpl(double position);
+    
+    /**
+    * The implementation of getPosition.
+    *
+    * @return    a double indicating the position
+    * @see       #getPosition()
+    */
+    abstract protected double getPositionImpl();
+
+    /**
+    * The implementation of moveTo.
+    *
+    * @param position   a double indicating the position to move to
+    * @return           nothing
+    * @see              #moveTo()
+    */
+    abstract protected void moveToImpl(double position);
+
+    /**
+    * Gets the wheel diameter in feet.
+    *
+    * @return   the wheel diameter in feet
+    */
+    abstract protected double getWheelDiameter();
+
+    /**
+    * Accelerates the mover towards the given speed.
+    * @param speed    a double indicating the speed to accelerate to
+    * @param maxStep  a double indicating the maximum speed change
+    * @return         nothing
+    * @see            #accelTowardImpl()
+    */
+    public final void accelToward(double speed, double maxStep) {
+        if (!isInitialized)
+            throw new IllegalStateException(
+                "Motor must be initialized first"
+            );
+        accelTowardImpl(speed, maxStep);
+    }
+
+    /**
+    * The implementation of accelToward.
+    * @param speed    a double indicating the speed to accelerate to
+    * @param maxStep  a double indicating the maximum speed change
+    * @return         nothing
+    * @see            #accelToward()
+    */
+    protected void accelTowardImpl(double speed, double maxStep) {
+        double currentSpeed = getSpeed();
+        double newSpeed = speed;
+        if (speed > currentSpeed) {
+            newSpeed = Math.min(currentSpeed + maxStep, speed);
+        } else {
+            newSpeed = Math.max(currentSpeed - maxStep, speed);
+        }
+        setSpeed(newSpeed);
```

Comment:
Do you foresee this being called many times or just once?

Reply:
I foresee it being called in `periodic`.